src/node/historical_queries_adapter.h

```diff
@@ -17,9 +17,131 @@ namespace ccf::historical
     kv::Consensus::View view,
     kv::SeqNo seqno)>;
 
+  using HandleHistoricalQueryWithReceipt = std::function<void(
+    ccf::EndpointContext& args,
+    StorePtr store,
+    TxReceiptPtr,
+    kv::Consensus::View view,
+    kv::SeqNo seqno)>;
+
 // Unused in most sample apps
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wunused-function"
+  static ccf::EndpointFunction receipt_adapter(
+    const HandleHistoricalQueryWithReceipt& f,
+    AbstractStateCache& state_cache,
+    const CheckAvailability& available)
+  {
+    return [f, &state_cache, available](EndpointContext& args) {
+      // Extract the requested transaction ID
+      kv::Consensus::View target_view;
+      kv::SeqNo target_seqno;
+
+      {
+        const auto target_view_opt =
```

Comment:
Most of these checks are repeated in the other historical adapter. Could we unify those?

Reply:
Hopefully, I started out by templatizing it, but I got into a bit an error rabbit hole, so I just forked it. I'll have another look now that everything is working and I'm cleaning up.

Reply:
Discussed with @eddyashton, merged with a single StatePtr interface.